axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/IndexManager.java

```diff
@@ -96,7 +96,11 @@ public void createIndex(Long segment, Map<String, SortedSet<PositionInfo>> posit
         RuntimeException lastError = new RuntimeException();
         for (int retry = 0; retry < 3; retry++) {
             try {
-                return getIndex(segment).getPositions(aggregateId);
+                Index idx = getIndex(segment);
```

Comment:
I would change getIndex to throw the exception instead of changing it only here. validIndex method would still work when it would receive an exception.